pymc3/stats.py

```diff
@@ -581,8 +584,8 @@ def add_warns(*args):
         warnings.filterwarnings('always')
 
         ics = []
-        for c, (t, m) in enumerate(zip(traces, models)):
-            ics.append((c, ic_func(t, m, pointwise=True)))
+        for n, (m, t) in zip(names, model_dict.items()):
+            ics.append((n, ic_func(t, m, pointwise=True)))
```

Comment:
in order to capture the warnings you need to increase by one the value of variable `c` . something like `c += 1`

Reply:
Notice that in the *model_comparison* notebook when you run waic for the pooled and hierarchical models you don't get a warning, but the result of `compare` indicates a warning (hierarchical model).

Reply:
OK, I reverted this